src/poetry/installation/executor.py

```diff
@@ -158,11 +160,39 @@ def execute(self, operations: list[Operation]) -> int:
                     operation.package.develop
                     and operation.package.source_type in {"directory", "git"}
                 )
-                if not operation.skipped and is_parallel_unsafe:
+
+                if operation.skipped:
+                    # Skipped operations are safe to execute in parallel
+                    tasks.append(
+                        self._executor.submit(self._execute_operation, operation)
+                    )
+                elif is_parallel_unsafe:
                     serial_operations.append(operation)
-                    continue
+                elif operation.package.source_type == "git":
+                    # Git operations on the same repository should be executed serially
+                    serial_git_operations[operation.package.source_url].append(
+                        operation
+                    )
+                else:
+                    tasks.append(
+                        self._executor.submit(self._execute_operation, operation)
+                    )
```

Comment:
Something feels weird here - the first `if` and final `else` clauses have identical code:
```python
if operation.skipped:
  tasks.append(self._executor.submit())
<snip bunch of elif clauses>
else:
  tasks.append(self._executor.submit())
```
Maybe it would make sense to fold the `operation.skipped` into `is_parallel_unsafe`? Like so, simplest I can think of:
```python
is_parallel_unsafe = operation.job_type == "uninstall" or (
    operation.package.develop
    and operation.package.source_type in {"directory", "git"}
)
# Skipped operations are safe to execute in parallel
if operation.skipped:
  is_parallel_unsafe = False
```
Then the if/else becomes simpler:
```python
if is_parallel_unsafe:
  # run serially
elif operation.package.source_type == "git":
  # Git operations on the same repository should be executed serially
else:
  # execute in parallel
```

Reply:
Agreed, nice suggestion!